fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -279,35 +279,41 @@ struct MetaclusterManagementWorkload : TestWorkload {
 		state bool forceJoin = deterministicRandom()->coinflip();
 
 		state std::vector<std::string> messages;
-		state bool retried = false;
-		loop {
-			try {
-				Future<Void> restoreFuture =
-				    MetaclusterAPI::restoreCluster(self->managementDb,
-				                                   clusterName,
-				                                   dataDb->db->getConnectionRecord()->getConnectionString(),
-				                                   ApplyManagementClusterUpdates::True,
-				                                   RestoreDryRun(dryRun),
-				                                   ForceJoinNewMetacluster(forceJoin),
-				                                   &messages);
-				Optional<Void> result = wait(timeout(restoreFuture, deterministicRandom()->randomInt(1, 30)));
-				if (result.present()) {
-					break;
-				}
-				retried = true;
-			} catch (Error& e) {
-				if (e.code() == error_code_conflicting_restore) {
-					ASSERT(retried);
-					CODE_PROBE(true, "MetaclusterManagementWorkload: timed out restore conflicts with retried restore");
-					continue;
-				} else if (e.code() == error_code_cluster_not_found) {
-					ASSERT(!dataDb->registered);
-					return Void();
+		try {
+			state bool retried = false;
+			loop {
+				try {
+					Future<Void> restoreFuture =
+					    MetaclusterAPI::restoreCluster(self->managementDb,
+					                                   clusterName,
+					                                   dataDb->db->getConnectionRecord()->getConnectionString(),
+					                                   ApplyManagementClusterUpdates::True,
+					                                   RestoreDryRun(dryRun),
+					                                   ForceJoinNewMetacluster(forceJoin),
+					                                   &messages);
+					Optional<Void> result = wait(timeout(restoreFuture, deterministicRandom()->randomInt(1, 30)));
+					if (result.present()) {
+						break;
+					}
+					retried = true;
+				} catch (Error& e) {
+					if (e.code() == error_code_conflicting_restore) {
+						ASSERT(retried);
+						CODE_PROBE(true,
+						           "MetaclusterManagementWorkload: timed out restore conflicts with retried restore");
+						continue;
+					}
+					throw;
```

Comment:
I believe we can take the existing code from this merge as is, since it is an implementation of the same bug fix you are doing here.